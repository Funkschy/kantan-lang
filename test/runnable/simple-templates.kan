import "io";

type S struct {}

def (s: S) print() {
    io.printf("S\n");
}

type T struct {}

def (t: T) print() {
    io.printf("T\n");
}

def [T] f(t: T) {
    t.print();
}

def [T] min(a: T, b: T): T {
    if a < b {
        return a;
    }
    return b;
}

def main() {
    io.printf("%d\n", min(1, 2));
    io.printf("%d\n", min(2, 1));

    f(S{});
    f(T{});
}
