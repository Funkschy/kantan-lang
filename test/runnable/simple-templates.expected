1
1
S
T
